source/Halibut/Transport/TcpConnectionFactory.cs

```diff
@@ -52,16 +54,22 @@ public async Task<IConnection> EstablishNewConnectionAsync(ExchangeProtocolBuild
             var certificateValidator = new ClientCertificateValidator(serviceEndpoint);
             var client = await CreateConnectedTcpClientAsync(serviceEndpoint, log, cancellationToken);
             log.Write(EventType.Diagnostic, $"Connection established to {client.Client.RemoteEndPoint} for {serviceEndpoint.BaseUri}");
-
-            var stream = client.GetStream();
+            
+            var networkStream = client.GetStream();
```

Comment:
Should this networkstream always be wrapped in a networkTimeoutStream? Are we wrapping the ssl stream at a later point?

Reply:
We could. But the regular 'sync' version respects the timeouts. So there's no need.

Do you think it worth doing anyway for consistency?

Reply:
After a chat, we decided to do it. Mainly because it means that the subsequent SSL methods will also respect the timeouts.